app/assets/v2/js/grants/funding.js

```diff
@@ -0,0 +1,198 @@
+            updateCartItem(grant.grant_id, 'grant_donation_amount', newAmount);
+        });
+
+        // Select appropriate currency
+        $(`#side-cart-currency-${grant.grant_id}`).val(grant.grant_donation_currency);
+
+        // Register currency change handler
+        $(`#side-cart-currency-${grant.grant_id}`).change(function() {
+            updateCartItem(grant.grant_id, 'grant_donation_currency', $(this).val());
+        });
+    });
+
+    toggleSideCart();
+}
+
+function hideSideCart() {
+    const isShowing = $('#side-cart').hasClass('col-3');
+
+    if (!isShowing) {
+        return;
+    }
+
+    toggleSideCart();
+
+    // Remove elements in cart
+    $("#side-cart-data")
+        .find("div.side-cart-row")
+        .remove();
+}
+
+function toggleSideCart() {
+    $('#side-cart').toggle();
+    $('#grants-details').toggleClass('col-12');
+    $('#grants-details').toggleClass('col-9');
+    $('#side-cart').toggleClass("col-3");
+    $('#side-cart').toggleClass("col-0");
+    $('#funding-card').toggleClass("mr-md-5");
+    $('#funding-card').toggleClass("mr-md-3");
+}
+
+function objectifySerialized(data) {
+    let objectData = {};
+
+    for (let i = 0; i < data.length; i++) {
+        const item = data[i];
+        objectData[item.name] = item.value;
+    }
+
+    return objectData;
+}
+
+function cartContainsGrantWithId(grantId) {
+    const cart = loadCart();
+    const idList = cart.map(grant => {
+        return grant.grant_id;
+    });
+
+    return idList.includes(grantId);
+}
+
+function addToCart(grantData) {
+    if (cartContainsGrantWithId(grantData.grant_id)) {
+        return;
+    }
+
+    // Add donation defaults
+    grantData.grant_donation_amount = 1;
+    grantData.grant_donation_currency = 'DAI';
+    grantData.grant_donation_num_rounds = 1;
+    grantData.grant_donation_clr_match = 250;
+
+    let cartList = loadCart()
+    cartList.push(grantData);
+    setCart(cartList);
+}
+
+function removeIdFromCart(grantId) {
+    let cartList = loadCart();
+
+    const newList = cartList.filter(grant => {
+        return (grant.grant_id !== grantId);
+    });
+
+    setCart(newList);
+}
+
+function updateCartItem(grantId, field, value) {
+    let cartList = loadCart();
+
+    let grant = null;
+
+    for (let index = 0; index < cartList.length; index++) {
+        const maybeGrant = cartList[index];
+
+        if (maybeGrant.grant_id === grantId) {
+            grant = maybeGrant;
+            break;
+        }
+    }
+
+    if (null === grant) {
+        throw new Error(`Tried to update grant with Id ${grantId} that is not in cart`);
+    }
+
+    grant[field] = value;
+
+    setCart(cartList);
+}
+
+function loadCart() {
+    let cartList = localStorage.getItem('grants_cart');
+
+    if (!cartList) {
+        return [];
+    }
+
+    return JSON.parse(cartList);
+}
+
+function setCart(list) {
+    localStorage.setItem('grants_cart', JSON.stringify(list));
+}
```

Comment:
^ would it make more sense to turn these all into Vue methods ?

Reply:
hey @thelostone-mc 👋. Hmm, yes that's a good thought. I agree we could probably use a better abstraction here. Thinking about the way `localStorage` cart data is shared between the Vue cart component and the jQuery code on the details page, it probably makes sense to make these static methods on a shared helper class or something along those lines.